src/components/Media/index.js

```diff
@@ -85,10 +89,17 @@ export default class Media extends PureComponent {
     ])
   }
 
+  onLoad = () => {
```

Comment:
you don't need onLoad here. This is controlled component it gets all state outside. You can use `props.icon === loaded`